### PR TITLE
Tests for timelock - check lock time verify (CLTV)

### PR DESCRIPTION
**Pull request by talelbaz**
The first part of #1257
Tests for timelock - CLTV (conditioned by time and block height).
The second part of #1257 will include tests for CSV and will be in a separate pull request.

### CODE REVIEW

File: domain/consensus/timelock_CLTV_test.go [hunk 1]
```diff
@@ -141,17 +136,108 @@ func TestCheckLockTimeVerifyConditionedByAbsoluteTime(t *testing.T) {
...
+		}
+		if !blockWithWrongLockTimeStatus.Equal(externalapi.StatusDisqualifiedFromChain) {
+			t.Fatalf("The status of blockWithWrongLockTime should be: %v, but got: %v", externalapi.StatusDisqualifiedFromChain,
+				blockWithWrongLockTimeStatus)
+		}
+	})
+}
+
+// TestCheckLockTimeVerifyConditionedByAbsoluteTime verifies that an output locked by the CLTV script has spendable only after
+//// the time is reached to the set target (compared to the past median time).
```
Comment (someone235) [inline domain/consensus/timelock_CLTV_test.go:215]: is -> has.
You also have extra `//` on the comment

File: domain/consensus/timelock_CLTV_test.go [hunk 2]
```diff
@@ -242,9 +329,131 @@ func TestCheckLockTimeVerifyConditionedByAbsoluteTime(t *testing.T) {
...
+			ScriptPublicKey: &externalapi.ScriptPublicKey{
+				Script:  nil,
+				Version: 0,
+			},
+		}
+		var tipHash *externalapi.DomainHash
+		timeStampBlockE := blockE.Header.TimeInMilliseconds()
+		stagingArea := model.NewStagingArea()
+		// Make sure the time limitation has passed.
+		for i := int64(0); ; i++ {
```
Comment (someone235) [inline domain/consensus/timelock_CLTV_test.go:414]: Why is it needed here and not in `TestCheckLockTimeVerifyConditionedByAbsoluteTime`?

**Review by someone235 — CHANGES_REQUESTED** [review]

File: domain/consensus/timelock_CLTV_test.go (hunk 2 above)
Comment (talelbaz) [reply domain/consensus/timelock_CLTV_test.go:414]: Needed in both places.
Already exists there.

File: domain/consensus/utils/txscript/scriptbuilder.go [hunk 3]
```diff
@@ -251,6 +251,36 @@ func (b *ScriptBuilder) AddInt64(val int64) *ScriptBuilder {
...
+		return b
+	}
+	lockTimeBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(lockTimeBytes, lockTime)
+	fixedSize := 7
+	for lockTimeBytes[fixedSize] == 0 {
+		fixedSize--
+	}
+	fixedLockTimeBytes := lockTimeBytes[:fixedSize+1]
+	return b.addData(fixedLockTimeBytes)
```
Comment (someone235) [inline domain/consensus/utils/txscript/scriptbuilder.go:266]: Why don't you call AddData?

File: domain/consensus/utils/txscript/scriptbuilder.go [hunk 4]
```diff
@@ -251,6 +251,36 @@ func (b *ScriptBuilder) AddInt64(val int64) *ScriptBuilder {
...
 }
 
+// AddLockTimeNumber gets a uint64 lockTime, converts it to byte array in little-endian, and then used the AddData function.
+func (b *ScriptBuilder) AddLockTimeNumber(lockTime uint64) *ScriptBuilder {
+	if b.err != nil {
+		return b
+	}
+	lockTimeBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(lockTimeBytes, lockTime)
+	fixedSize := 7
```
Comment (someone235) [inline domain/consensus/utils/txscript/scriptbuilder.go:261]: This is index, not size

**Review by someone235 — CHANGES_REQUESTED** [review]

**Review by someone235 — DISMISSED** [review]

**Review by someone235 — APPROVED** [review]

**Pull request merged**